### PR TITLE
Prometheus: make parsing of Infinity variants case-insensitive

### PR DESCRIPTION
**Pull request by leeoniya**
Fixes https://github.com/grafana/support-escalations/issues/2453

(probably)

cc @taleena

### CODE REVIEW

**Comment by gabor** [comment]
@leeoniya i'm not working on prometheus-code anymore, but if i'm not mistaken, this code-path is only used in "browser-mode" these days, and that one is deprecated, sorry 😿  (pinging @toddtreece to doublecheck what i wrote)

the normal flow of code parses the result in the go-code, which uses https://github.com/prometheus/client_golang to transform the JSON data to go-data-structures (in our case to a floating-point-number).

i don't know if the https://github.com/prometheus/client_golang code handles all the other infinity-strings, but if they do, then this problem is not happening in current-grafana-version (at least not in non-browser-mode).

File: public/app/plugins/datasource/prometheus/result_transformer.ts [hunk 1]
```diff
@@ -612,12 +612,8 @@ function sortSeriesByLabel(s1: DataFrame, s2: DataFrame): number {
...
 
 function parseSampleValue(value: string): number {
-  switch (value) {
-    case POSITIVE_INFINITY_SAMPLE_VALUE:
-      return Number.POSITIVE_INFINITY;
-    case NEGATIVE_INFINITY_SAMPLE_VALUE:
-      return Number.NEGATIVE_INFINITY;
-    default:
-      return parseFloat(value);
+  if (INFINITY_SAMPLE_REGEX.test(value)) {
```
Comment (toddtreece) [inline public/app/plugins/datasource/prometheus/result_transformer.ts:615]: is `parseSampleValue` run for every value in the response? if so, i'm wondering if there's a significant performance hit with switching to regex matching

Comment (leeoniya) [reply public/app/plugins/datasource/prometheus/result_transformer.ts:615]: i have an entire comment and bench dedicated to answering this (scroll up) :rofl:

Comment (toddtreece) [reply public/app/plugins/datasource/prometheus/result_transformer.ts:615]: 🤦‍♂️ my bad. i attempted to quickly scan this while my toddler was busy with a book

**Review by toddtreece — APPROVED** [review]
the issue probably needs to be tested against the backend version of the data source since it's a different code path as @gabor mentioned, but this looks good to me for the browser/proxy mode queries 👍

**Comment by toddtreece** [comment]
after talking with @leeoniya, it looks like this will also solve the problem for backend responses. the original tests were using `transform`, but `transformV2` uses `parseSampleValue` to parse label values

**Pull request merged**